src/features/session_replay/instrument/index.js

```diff
@@ -44,10 +54,15 @@ export class Instrument extends InstrumentBase {
   }
 
   async #startRecording (mode) {
+    if (this.featAggregate) {
+      if (this.featAggregate.entitled) this.featAggregate.initializeRecording(mode, true)
```

Comment:
#apiRestartRecording should:

1. check if the agg is imported.
  a. if so, check if it's currently recording and if so, done
  b. if not recording, start recording if entitled to do so, done. If not entitled, done.
2. check if a startup is progress. if so, done. if not, start it up. ( aka just call `#startRecording`)

It's close to what you have here, except for the bit about checking for recording in-progress, but easier to read & analyze if preload start doesn't deal with step 1 logic.

Reply:
One more thing here. `mode` has to be a class instance variable that's changeable because of async. Consider if preload startup with session-inherited ERROR mode, but then api recordReplay is called. A startup is already in progress, but that startup will end up recording using ERROR instead of FULL as the wanted behavior in this case. Therefore, the recording should use whatever is the latest mode at the time.
I.e. `#apiRestartRecording` will set the local mode to be FULL, and _then_ quit to let the in-progress startup finish.

Reply:
```The context here is you don't know if replay was stopped/paused/unstarted to begin with. So this shouldn't initializeRecording potentially again```

we should talk through this to get everything aligned, because initializeRecording checks for `.blocked` which would be the only case where we would not want to start recording again.